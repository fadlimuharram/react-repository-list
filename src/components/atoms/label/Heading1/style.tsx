import styled from 'styled-components';

export const Wrapper = styled.h1`
	font-family: Montserrat, sans-serif;
	font-style: normal;
	font-weight: 600;
	font-size: 1.75rem;
	line-height: 34px;
	word-wrap: break-word;
	text-align: center;
`;
