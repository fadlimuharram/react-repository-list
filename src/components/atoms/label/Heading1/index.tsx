import React from 'react';
import { Wrapper } from './style';

const Heading1: React.FC = ({ children }) => {
	return <Wrapper data-testid="heading-1">{children}</Wrapper>;
};

export default Heading1;
