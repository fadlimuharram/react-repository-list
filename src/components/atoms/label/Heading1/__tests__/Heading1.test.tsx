import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Heading1 from '..';

afterEach(() => {
	cleanup();
});

test('should render Hello from heading 1', () => {
	render(<Heading1>Hello</Heading1>);
	const headingElm = screen.getByTestId('heading-1');
	expect(headingElm).toBeInTheDocument();
	expect(headingElm).toHaveTextContent('Hello');
});

test('Match Heading1 Snapshot', () => {
	const tree = renderer.create(<Heading1>Hello</Heading1>).toJSON();

	expect(tree).toMatchSnapshot();
});
