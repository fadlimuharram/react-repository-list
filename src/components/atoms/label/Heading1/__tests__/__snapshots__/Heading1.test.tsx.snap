// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Heading1 Snapshot 1`] = `
<h1
  className="sc-bdnxRM fxAJjl"
  data-testid="heading-1"
>
  Hello
</h1>
`;
